cli/ops/fs.rs

```diff
@@ -84,18 +99,29 @@ fn op_open(
       state.check_write(&path)?;
     }
 
+    if gave_mode && !(options.create || options.create_new) {
+      return Err(OpError::type_error(
+        "specified mode without allowing file creation".to_string(),
+      ));
```

Comment:
This error handling seems unnecessary. Rust will raise an the error.

Reply:
If I remove this check, Rust raises no error for `Deno.openSync(existing_filepath, { create:false, createNew:false, mode:0o600, read:true, write:true});`. See below.

Reply:
If Rust doesn't raise, why do we need to?